app/uk/gov/hmrc/gform/graph/Recalculation.scala

```diff
@@ -88,6 +89,14 @@ class Recalculation[F[_]: Monad, E](
         noStateChange(res)
       }
 
+      def matchRegex(formCtx: FormCtx, regex: Regex): Boolean = {
+        val typeInfo1 = formModel.toFirstOperandTypeInfo(formCtx)
```

Comment:
As we know we are matching RegEx, which is a string. Do we need to determined type first? Can we just convert whatever the value is to string and evaluate?

Reply:
We can match numbers as well. But more importantly we need to utilise the heavy work done by `ExpressionResult` since in `includeIf` we may be matching against invisible field.